src/QueryReflection/DibiQueryReflector.php

```diff
@@ -0,0 +1,126 @@
+<?php
+
+declare(strict_types=1);
+
+namespace staabm\PHPStanDba\QueryReflection;
+
+use Dibi\Connection;
+use Dibi\ConstraintViolationException;
+use PHPStan\Type\Constant\ConstantArrayTypeBuilder;
+use PHPStan\Type\Constant\ConstantStringType;
+use PHPStan\Type\MixedType;
+use PHPStan\Type\Type;
+use staabm\PHPStanDba\Error;
+use staabm\PHPStanDba\TypeMapping\MysqliTypeMapper;
+
+final class DibiQueryReflector implements QueryReflector, RecordingReflector
```

Comment:
I don't think we need this reflector, in case we could get the underlying connection out of a dibi-connection

Reply:
we can (via `$connection->getDriver()->getResource()`) but what about the way I replace placeholders in `DibiQueryReflector::simulateQuery`? Also the `fetchtypes` parameter in `getResultType` is irrelevant for dibi (not even sure what it is)